manager/src/main/java/com/redhat/service/smartevents/manager/connectors/ConnectorsApiClientImpl.java

```diff
@@ -67,15 +67,18 @@ public Connector getConnector(String connectorExternalId) {
         ConnectorsApi connectorsAPI = createConnectorsAPI();
 
         try {
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug(String.format("Retrieving Connector with ID '%s'", connectorExternalId));
-            }
+            LOGGER.debug("Retrieving Connector with ID '{}'", connectorExternalId);
             return connectorsAPI.getConnector(connectorExternalId);
         } catch (ApiException e) {
-            if (e.getCode() == HttpStatus.SC_NOT_FOUND) {
-                return null;
+            switch (e.getCode()) {
+                case HttpStatus.SC_NOT_FOUND:
+                case HttpStatus.SC_GONE:
+                    LOGGER.info("Connector could not be found in MC Fleet Manager.");
+                    return null;
+                default:
+                    LOGGER.error("Failed to retrieve connector: {}", e.getMessage());
+                    throw new ConnectorGetException("Error while retrieving the connector on MC Fleet Manager", e);
```

Comment:
@manstis Same point here for the `Exception` message. If the `Exception` message is:

```
String message = String.format("Failed to retrieve Connector with id '%s', from Connector Namespace '%s'", connectorExternalId, mcNamespaceId);
throw new ConnectorGetException(message, e)
```
Then we don't need the additional `LOGGER.error` here IMHO. It will be caught and logged in the `ConnectorWorker` giving us a single statement in the log, with the chained stacktrace.

Reply:
Done.

Reply:
@sutaakar I'm not sure @robpblake opinion is correct.

From what I saw on [MGDOBR-634](https://issues.redhat.com/browse/MGDOBR-634) we only get the message from `ConnectorGetException` and the underlying cause is not recorded. I think I should either log the original error here or add `e.printStackTrace()` to `AbstractWorker` exception handler. WDYT?

Reply:
@manstis I think the log statement in `AbstractWorker` needs to change from:

```
LOGGER.info("Failed to delete dependencies for '{}' [{}].\n"
                        + "Work status: {}\n"
                        + "{}",
                        managedResource.getName(),
                        managedResource.getId(),
                        work,
                        e.getMessage());
```

to 

```
LOGGER.info("Failed to delete dependencies for '{}' [{}].\n"
                        + "Work status: {}\n"
                        + "{}",
                        managedResource.getName(),
                        managedResource.getId(),
                        work,
                        e);
```

Note the drop of the `e.getMessage()` call.

cheers,

Rob

Reply:
So the complete stacktrace with causes is printed in `AbstractWorker`? Fine for me.
So the API client will just wrap the error it gets from external system and pass it above.

Reply:
Thanks @robpblake I will find a rock to lay under until the day passes. Update made.